Lib/imaplib.py

```diff
@@ -1579,9 +1579,10 @@ def run(cmd, args):
             run(cmd, args)
 
         for ml in run('list', ('/tmp/', 'yy%')):
-            mo = re.match(r'.*"([^"]+)"$', ml)
-            if mo: path = mo.group(1)
-            else: path = ml.split()[-1]
+            if (mo := re.match(r'.*"([^"]+)"$', ml)):
+                path = mo.group(1)
+            else:
+                path = ml.split()[-1]
             run('delete', (path,))
 
```

Comment:
Note that the original author was already doing 

if: then_stuff
else: else_stuff

to keep it from taking up too much white space.  Your rewrite is actually a line longer, but only because you've gone back to the standard "suite indented on the next line" format.  So that suggests this is relieving a real problem.